feat: [CDS-80915]: Update MultiTypeMap and MultiTypeMapInputSet to support default/allowed values

rameshx opened the pull request:
### Summary

This updates MultiTypeMap and MultiTypeMapInputSet components to add default value and allowed values support to the value fields of `envVariables` in the Run step.

#### Screenshots

https://github.com/harness/harness-core-ui/assets/52284933/b80b756c-3d18-4f0a-a82c-acb3afac28e2


#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [x] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

chetanvekariya-harness commented on src/modules/70-pipeline/components/InputSetView/MultiTypeMapInputSet/__test__/MultiTypeMapInputSet.test.tsx:
```diff
@@ -0,0 +1,158 @@
+      fireEvent.click(getByTestId('add-test'))
+    })
+
+    expect(queryByNameAttribute('test[0].key', container)).toBeTruthy()
```
Can we use toBeInTheDocument instead? I think this could return `null` and `null` is truthy.

rameshx replied:
Marking as resolved as this is an existing test and `null` is always falsy.

chetanvekariya-harness commented on src/modules/70-pipeline/components/InputSetView/MultiTypeMapInputSet/__test__/MultiTypeMapInputSet.test.tsx:
```diff
@@ -0,0 +1,158 @@
+
+  test('Should render properly', () => {
+    const { container } = render(<TestComponent initialValues={{}} />)
+    expect(container).toMatchSnapshot()
```
Can we add assertions instead of multiple snapshots?

rameshx replied:
Marking as resolved as this is an existing test

d1wilko reviewed (approved):
No issues stand out :)

chris-harness commented:
I notice in the video that the size of the "Key" input box in the step form changes when the input type is changed. The vertical spacing between environment variables also seems wider than it used to be on that form, and feels a bit excessive.

See timestamp 0:16 in the screen recording for examples of both issues.

chris-harness reviewed (changes requested):
When I run a step using the "input" mode for an env var, with validation configured, the value placed in the environment variable incorrectly includes the validation description. The CI Pod gets `Critical.allowedValues(0,Critical,High,Medium,Low,Info)` in the env var when it should just receive the selected value `Critical`

rameshx commented:
> I notice in the video that the size of the "Key" input box in the step form changes when the input type is changed. The vertical spacing between environment variables also seems wider than it used to be on that form, and feels a bit excessive.
> 
> See timestamp 0:16 in the screen recording for examples of both issues.

Refactored all styles of the component to use grid. Updated the video.

rameshx commented:
> When I run a step using the "input" mode for an env var, with validation configured, the value placed in the environment variable incorrectly includes the validation description. The CI Pod gets `Critical.allowedValues(0,Critical,High,Medium,Low,Info)` in the env var when it should just receive the selected value `Critical`

Thanks for testing this e2e. Could you share the execution link?

chris-harness commented:
> > When I run a step using the "input" mode for an env var, with validation configured, the value placed in the environment variable incorrectly includes the validation description. The CI Pod gets `Critical.allowedValues(0,Critical,High,Medium,Low,Info)` in the env var when it should just receive the selected value `Critical`
> 
> Thanks for testing this e2e. Could you share the execution link?

I ran it locally, with QA env as the backend. I suspect you don't have access to the account, though: https://localhost:8181/ng/account/BdsgiWzwT7CQFeJl9XkQ3A/all/orgs/default/projects/STO/pipelines/Chris_Demo/executions/J9xAOATaTYCjt5jOgdu7Ug/pipeline?storeType=INLINE&stage=4RqL1ioARS--rHhVV0oefQ

Here's the YAML for the step:
```yaml
                  - step:
                      type: Security
                      name: OWASP 1
                      identifier: OWASP_1
                      spec:
                        privileged: true
                        settings:
                          policy_type: orchestratedScan
                          scan_type: repository
                          repository_project: chris_nodegoat
                          repository_branch: <+codebase.branch>
                          product_name: owasp
                          product_config_name: owasp
                          sto_dump_runner_logs: "true"
                          sto_plugin_normalization: "true"
                          fail_on_severity: <+input>.default(0).allowedValues(0,Critical,High,Medium,Low,Info)
                        imagePullPolicy: Always
                      when:
                        stageStatus: Success
```

kalyansai-harness commented:
@chetanvekariya-harness @d1wilko could you please review the PR soon.

chetanvekariya-harness commented on src/modules/70-pipeline/components/InputSetView/MultiTypeMapInputSet/ValueInputSet.tsx:
```diff
@@ -0,0 +1,45 @@
+  return <Select {...selectProps} />
+}
+
+export function ValueInputSet(props: ValueInputSetProps): JSX.Element {
```
nit: MultiTypeMapValueInputSet would be more suitable component name

d1wilko reviewed (approved):
LGTM
